src/views/Profile/AdminTable/AdminDatePicker.tsx

```diff
@@ -32,18 +35,100 @@ interface IMaterialsDate {
   setChanges: (...params: any[]) => void;
 }
 
+interface ISelectedDay {
+  position: string;
+  bgColor: string;
+  value: string;
+}
+
+const datePickerReducer = (
+  state: {
+    firstSelectedDay: ISelectedDay;
+    secondSelectedDay: ISelectedDay;
+  },
+  action: { type: string; payload: ISelectedDay },
+): {
+  firstSelectedDay: ISelectedDay;
+  secondSelectedDay: ISelectedDay;
+} => {
+  switch (action.type) {
+    case 'SET_SELECTED_DAY':
```

Comment:
instead of using generic `SET_SELECTED_DAY` type, I propose to have separate `SET_START_DAY` and `SET_END_DAY` right from the start. This give us 2 benefits: 
1. better readability
a. (no `if .. ` nested into `switch/case` - the less nesting the easier to understand) 
b. explicit types for `START` and `END` instead of passing this as a part of payload
2, more flexibility: maybe it has not been stated yet, but typically it's expected that when we set end date before current start date or in opposite - setting start date after current end date, we either clear or update counterpart(e.g. when setting start date before current end date we either clear end date at all or update it to be equal to start date); even if this was not required yet, extra logic is still easier to apply with separate action types